### PR TITLE
Add functionality to generate input type fields' default values in service generation

### PR DESCRIPTION
**Pull request by chathuranga-jayanath-99**
## Purpose
$title

### CODE REVIEW

File: graphql-cli/src/test/resources/serviceGen/graphqlSchemas/valid/SchemaWithInputTypeDefaultParametersApi.graphql [hunk 1]
```diff
@@ -0,0 +1,19 @@
+type Query {
+    book(id: Int!, title: String): Book
```
Comment (ThisaruGuruge) [inline graphql-cli/src/test/resources/serviceGen/graphqlSchemas/valid/SchemaWithInputTypeDefaultParametersApi.graphql:2]: How do we handle default values on primitive inputs? 

```graphql
book(id: Int! = 1, title: String): Book
```

Shall we have a test for that too?

**Review by ThisaruGuruge — COMMENTED** [review]
Looks like there are a few lines that aren't covered in code coverage. Shall we at least try to cover basic error scenarios?

File: graphql-cli/src/test/resources/serviceGen/graphqlSchemas/valid/SchemaWithInputTypeDefaultParametersApi.graphql (hunk 1 above)
Comment (chathuranga-jayanath-99) [reply graphql-cli/src/test/resources/serviceGen/graphqlSchemas/valid/SchemaWithInputTypeDefaultParametersApi.graphql:2]: I have added tests to handle these scenarios in the previous pull request.

Comment (ThisaruGuruge) [reply graphql-cli/src/test/resources/serviceGen/graphqlSchemas/valid/SchemaWithInputTypeDefaultParametersApi.graphql:2]: My bad.

File: graphql-cli/src/test/java/io/ballerina/graphql/generator/ballerina/ServiceTypesGeneratorTest.java [hunk 2]
```diff
@@ -710,4 +762,30 @@ public void testGenerateSrcForSchemaWithFileUploadFields() {
             Assert.fail(e.getMessage());
         }
     }
+
+    @Test(groups = {"service-type-for-objects"}, description = "Test for schema with input type fields having default" +
+            " values")
```
Comment (ThisaruGuruge) [inline graphql-cli/src/test/java/io/ballerina/graphql/generator/ballerina/ServiceTypesGeneratorTest.java:767]:
```suggestion
    @Test(
        groups = {"service-type-for-objects"},
        description = "Test for schema with input type fields having default values")
```

File: graphql-cli/src/test/java/io/ballerina/graphql/generator/ballerina/ServiceTypesGeneratorTest.java [hunk 3]
```diff
@@ -205,14 +208,66 @@ public void testGenerateSrcForRecordForced() {
...
             String expectedServiceTypesContent = readContentWithFormat(expectedServiceTypesFile);
             String writtenServiceTypesContent = readContentWithFormat(this.tmpDir.resolve("types.bal"));
             Assert.assertEquals(expectedServiceTypesContent, writtenServiceTypesContent);
-        } catch (ValidationException | IOException | ServiceTypesGenerationException e) {
+        } catch (ValidationException | IOException | ServiceGenerationException e) {
+            Assert.fail(e.getMessage());
+        }
+    }
+
+    @Test(groups = {"record-type-for-objects"}, description = "Test for schema with object implementing interface")
```
Comment (ThisaruGuruge) [inline graphql-cli/src/test/java/io/ballerina/graphql/generator/ballerina/ServiceTypesGeneratorTest.java:223]: Check other places

```suggestion
    @Test(
        groups = {"record-type-for-objects"}, 
        description = "Test for schema with object implementing interface"
    )
```

File: graphql-cli/src/test/java/io/ballerina/graphql/generator/ballerina/ServiceTypesGeneratorTest.java [hunk 4]
```diff
@@ -413,21 +473,23 @@ public void testGenerateSrcForSchemaWithDocsInQueryResolverFunctions() {
...
         }
     }
 
 
-    @Test(groups = {"service-type-for-objects"},
-            description = "Test for schema with docs in arguments in resolver functions")
+    @Test(
+            groups = {"service-type-for-objects"},
+            description = "Test for schema with docs in arguments in resolver functions"
+    )
```
Comment (ThisaruGuruge) [inline graphql-cli/src/test/java/io/ballerina/graphql/generator/ballerina/ServiceTypesGeneratorTest.java:492]: Check indentation

```suggestion
    @Test(
        groups = {"service-type-for-objects"},
        description = "Test for schema with docs in arguments in resolver functions"
    )
```

File: graphql-cli/src/test/java/io/ballerina/graphql/generator/ballerina/ServiceTypesGeneratorTest.java [hunk 5]
```diff
@@ -699,15 +766,71 @@ public void testGenerateSrcForSchemaWithFileUploadFields() {
...
+            String writtenServiceTypesContent = readContentWithFormat(this.tmpDir.resolve("types.bal"));
+            Assert.assertEquals(expectedServiceTypesContent, writtenServiceTypesContent);
+        } catch (ValidationException | IOException | ServiceGenerationException e) {
+            Assert.fail(e.getMessage());
+        }
+    }
+
+    @Test(
+            groups = {"service-type-for-objects"},
+            description = "Test for schema with input type fields having default values"
```
Comment (ThisaruGuruge) [inline graphql-cli/src/test/java/io/ballerina/graphql/generator/ballerina/ServiceTypesGeneratorTest.java:783]: I don't think these descriptions add much value. Function name is descriptive enough.

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**